integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonShowCacheCommand.scala

```diff
@@ -135,23 +193,46 @@ case class CarbonShowCacheCommand(tableIdentifier: Option[TableIdentifier],
       carbonTables.flatMap {
         mainTable =>
           try {
-            makeRows(getTableCacheFromIndexServer(mainTable)(sparkSession), mainTable)
+            val row = makeRows(getTableCacheFromIndexServer(showExecutorCache,
+              mainTable)(sparkSession), mainTable)
+            var res: List[Any] = null
+            for (i <- row.toList) {
+              res = i.toSeq.toList
+            }
+            if (res(1) == 0 && res(2) == 0 && res(3) == 0) {
+              Seq()
+            } else {
+              row
+            }
           } catch {
             case ex: UnsupportedOperationException => Seq()
           }
       }
-    } else { Seq() }
+    } else {
+      Seq()
+    }
 
     val driverRows = if (cache != null) {
       carbonTables.flatMap {
         carbonTable =>
           try {
-            makeRows(getTableCacheFromDriver(sparkSession, carbonTable), carbonTable)
+            val row = makeRows(getTableCacheFromDriver(sparkSession, carbonTable), carbonTable)
+            var res: List[Any] = null
+            for (i <- row.toList) {
+              res = i.toSeq.toList
```

Comment:
No need to convert to List. same can be done for Seq also

Reply:
done